### DISCUSSION TITLE
Nullity validation for Controller

### DISCUSSION THREAD

**Discussion by babyfish-ct** [babyfish-ct/jimmer#406] [Ideas]
In `0.8.86`, Jimmer validates the nullity of the parameters of RestController, including primitive types and there box types. Eg:
 
```java
public void f(@ReqeustParam Long id)
```
will cause an error, because `Long` means optional but `@RequestParam(required = true)` means mandatory.

In order to fix it, users have 2 choices:

1.  ```java
    public void f(@ReqeustParam long id) // `long` means mandatory
    ```
    
2.  ```java
    public void f(@ReqeustParam(required = false) Long id) // `required = false` means optional
    ```
    
Is it validation too strict? Would jimmer keep the validation for primitive/box types  or remove it?

**Comment by TokgoRonin** [comment]
我建议是不要动这一块，controller的入参，惯性太大了，此举会劝退不少想要尝试jimmer的新用户。如果非要做这个事情的话，我建议给一个配置项，默认关闭，想开启严格校验时，由用户手动开启配置项。

**Comment by sunshio** [comment]
默认关闭吧

**Comment by pot-mot** [comment]
和 database-validation 类似的整一个配置项，默认关闭
